app/core/project/project_params_builder.rb

```diff
@@ -0,0 +1,93 @@
+class ProjectParamsBuilder
+  attr_accessor :row
+  attr_reader :messages, :project
+
+  def initialize(account, *param_hash)
+    @row = param_hash
+    @editor_account = account
+    @forge_id = Forge.find_by(name: 'Github').id
+
+    @licenses = {}
+  end
+
+  def build_project
+    @messages = []
+    @kb_project_id = @row.values[0]
+    process_row
+  end
+
+  private
+
+  def process_row
+    add_row_params
+    validate_row
+    @project = create_project
+  end
+
+  def create_project
+    params = set_params
+    builder = ProjectBuilder.new(@editor_account, params)
+    @project = builder.create
+    raise ActiveRecord::RecordNotSaved, project.errors.messages unless @project.errors.messages.empty?
+    @messages << "KB project #{@kb_project_id} has been imported to Open Hub as project #{@project.id}"
+    @project
+  end
+
+  def validate_row
+    raise ProjectParamsError, "kb id:#{@kb_project_id} contains a blank description" if @row['description'].blank?
+    raise ProjectParamsError, "kb id:#{@kb_project_id} project already exists"  if @row['owner'].nil?
+  end
+
+  def add_row_params
+    @row['forge_id'] = @forge_id
+    @row['owner_at_forge'], @row['name_at_forge'] = split_name(@row['owner_name'])
+    @row['owner'], @row['project_name'] = set_project_name(@row['owner_name'])
+  end
+
+  def split_name(owner_name)
+    owner_name.split('/')
+  end
+
+  def set_project_name(owner_name)
+    owner, project = split_name(owner_name)
+    # check if project and owner already exists
+    return if Project.where(owner_at_forge: owner, name_at_forge: project).exists?
```

Comment:
Leaving the validate_row with the new Exception class.  I think it's clean to have a single method that handles the validation, rather than have the validations strewn around the code.